src/main/scala/formal/FormalUtils.scala

```diff
@@ -46,12 +46,13 @@ object Property {
 
   def apply(dir: MonitorDirection, cond: Bool, message: String,
             prop_type: PropertyClass=PropertyClass.Default,
-            idx: String = "")(implicit sourceInfo: SourceInfo): Unit = {
+            idx: String = "",
+            custom_name: String = "")(implicit sourceInfo: SourceInfo): Unit = {
     val line_info = sourceInfo match {
       case SourceLine(filename, line, col) => s"${filename}_L${line}_C${col}_I${idx}".replace('.', '_')
       case _ => ""
     }
-    val proposed_src = prop_type.toString + "_" + line_info
+    val proposed_src = if (custom_name == "") prop_type.toString + "_" + line_info else custom_name
```

Comment:
Line info seems to be helpful.  Can you still keep that somehow? (Maybe don't need it as part of a label, but add it to message or something).

Reply:
Why/where would you want the line info if the person who wrote the assertion specified a specific name and a specific message? Can't you just search for that specific name in the codebase?